src/main/java/com/hr/app/jpaexample/controller/DepartmentController.java

```diff
@@ -32,5 +29,12 @@ public Department getDepartmentById(@PathVariable Long id) {
         return departmentService.getDepartmentById(id);
     }
 
+    @PostMapping(value="/insert", consumes = "application/json", produces = "application/json")
+    public Department insertRow(@RequestBody Department department){
+
+        return departmentService.insertRow(department);
```

Comment:
Refer this API design document: https://blog.stoplight.io/crud-api-design

@PostMapping
public Department **createDepartment**(@RequestBody **DepartmentDto** departmentDto)

@PostMapping will have by default consumes and produce as application/json only

Reply:
1) insertRow is generic name, you need to have specific naming conventions should be used whatever you are doing should be self explanatory by your method name example: **createDepartment**
2) Dempartment we already have entity name we should not use same entity class for request object, instead you should create new **DepartmentRequest** and use Mapper to copy value from DepartmentRequest object to Demaprtment entity object. and in service class method name should be **saveDepartment**  or **createDepartment** not the insert row